### DISCUSSION TITLE
Security Fix: Object Navigation from the Lock Screen

### DISCUSSION THREAD

**Discussion by seanbudd** [nvaccess/nvda#13361] [General]
Use this discussion for any technical questions surrounding the security fix for object navigation from the lock screen. Please [responsibly disclose](https://en.wikipedia.org/wiki/Coordinated_vulnerability_disclosure)  potential security issues to [info@nvaccess.org](mailto:info@nvaccess.org).

2021.3.2 includes a fix for a Windows issue that allowed the use of object navigation to read content on the desktop from the lock screen. This affects Windows 10 and 11.
However, this requires physical access, so it is only a potential issue for locked computers in shared areas. 

Object navigation can only occur from the lock screen, not the secure sign-in screen where your password is entered. You can prevent this issue when using older NVDA versions by disabling the lock screen. Disabling the Windows lock screen will cause locking the computer to go straight to the secure sign-in screen. To do this:

1. Open the run dialog with Windows+R
2. Enter and run: _gpedit.msc_ (may require administrative access)
3. Using the “Local Group Policy Editor” window 

   - Navigate to _Local Computer Policy, Computer Configuration, Administrative Templates, Control Panel, Personalization, Do Not Display the Lock Screen_
   - Enable "Do Not Display the Lock Screen"

6. Confirm with Windows+L that the lock screen is skipped and Windows goes directly to the secure sign-on screen.

The details of this issue can be found in the pull request: [Prevent Object Navigation Outside of the Lock Screen #13328](https://github.com/nvaccess/nvda/pull/13328).

For further information, please contact NV Access via [info@nvaccess.org](mailto:info@nvaccess.org).

**Comment by stickbear2015** [comment]
note that if your running windows 10 home the steps outlined above will not work as gpedit.msc is not available.
I can not speak to windows 11.

**Comment by stickbear2015** [comment]
after some research, here are the steps for windows 10 home users. These steps are given without warranty or fitness for any particular purpose.
I am not responsible if following these steps makes your system unstable or otherwise unusable.
I've modified these steps slightly from their original source as they were originally written for asighted user to follow.
[the original source is here.](https://www.howtogeek.com/270493/how-to-disable-the-lock-screen-in-windows-10/#:~:text=Select%20%E2%80%9CAdministrative%20Templates%E2%80%9D%20and%20then,%E2%80%9D%20when%20you're%20done.)
Feel free to take these and add them to the original post .
for windows 10 home users, This setting can be changed in the Windows Registry. To open the Registry Editor, start by pressing the Windows + R keys to open the Run window. Type “regedit” and then press enter to activate OK.
> Warning: The Registry Editor is a very powerful tool that can make your system unstable or worse if you use it incorrectly. Understand there is some risk involved with Registry edits. If your not comfortable with doing changes like this, don't attempt them, and ask someone you trust for assistance.

Now that your in the registry editor:

* expand “HKEY_LOCAL_MACHINE.
* Now expand “SOFTWARE.”
* expand policies 
* expand “Microsoft” and then hit your applications key or shift+f10 on the “Windows” folder. Select New then Key from the menu.
* you'll get an edit field, type “Personalization” without the quotes, replacing `New Key #1`.
* Now, hit the applications key or shift+f10 on the “Personalization” folder we just created. * Select New then DWORD (32-bit) Value from the menu.
* in the edit box that appears, replace `New Value #1` with “NoLockScreen” without the quotes.
* Now, press tab to the list view, then press enter on “NoLockScreen” and enter “1” for the “Value Data” field. and hit enter to activate “OK” when you’re done.
